Bottom: 789ebf1ea9946721afcde07b6137929f4bb141ff
Top:    950e49c801dd19776c393783bda14e98bd201e49
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-23 22:27:19 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 2b7f2b4..0e161ee 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2460,7 +2460,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
